FirebaseSegmentation/Tests/Unit/SEGContentManagerTests.m

```diff
@@ -19,41 +19,63 @@
 #import "SEGNetworkManager.h"
 
 #import <FirebaseCore/FirebaseCore.h>
-#import <FirebaseInstanceID/FIRInstanceID.h>
+#import <FirebaseInstallations/FirebaseInstallations.h>
 #import <OCMock/OCMock.h>
+#import "FIRInstallationsIDController.h"
 
 @interface SEGContentManager (ForTest)
 - (instancetype)initWithDatabaseManager:databaseManager networkManager:networkManager;
 @end
 
-@interface FIRInstanceIDResult (ForTest)
-@property(nonatomic, readwrite) NSString *instanceID;
-@property(nonatomic, readwrite) NSString *token;
+@interface FIRInstallations (Tests)
+@property(nonatomic, readwrite, strong) FIROptions *appOptions;
+@property(nonatomic, readwrite, strong) NSString *appName;
+
+- (instancetype)initWithAppOptions:(FIROptions *)appOptions
+                           appName:(NSString *)appName
+         installationsIDController:(FIRInstallationsIDController *)installationsIDController
+                 prefetchAuthToken:(BOOL)prefetchAuthToken;
 @end
 
-@interface FIRInstanceID (ForTest)
-+ (instancetype)instanceIDForTests;
+@interface FIRInstallationsAuthTokenResult (ForTest)
+- (instancetype)initWithToken:(NSString *)token expirationDate:(NSDate *)expirationTime;
 @end
 
 @interface SEGContentManagerTests : XCTestCase
 @property(nonatomic) SEGContentManager *contentManager;
 @property(nonatomic) id instanceIDMock;
 @property(nonatomic) id networkManagerMock;
+@property(nonatomic) FIRInstallations *installationsMock;
+@property(nonatomic) id mockIDController;
+@property(nonatomic) FIROptions *appOptions;
+@property(readonly) NSString *firebaseAppName;
 
 @end
 
 @implementation SEGContentManagerTests
 
 - (void)setUp {
   // Setup FIRApp.
-  XCTAssertNoThrow([FIRApp configureWithOptions:[self FIRAppOptions]]);
+  self->_firebaseAppName = @"my-firebase-app-id";
+  XCTAssertNoThrow([FIRApp configureWithName:self.firebaseAppName options:[self FIRAppOptions]]);
   // TODO (mandard): Investigate replacing the partial mock with a class mock.
-  self.instanceIDMock = OCMPartialMock([FIRInstanceID instanceIDForTests]);
-  FIRInstanceIDResult *result = [[FIRInstanceIDResult alloc] init];
-  result.instanceID = @"test-instance-id";
-  result.token = @"test-instance-id-token";
-  OCMStub([self.instanceIDMock
-      instanceIDWithHandler:([OCMArg invokeBlockWithArgs:result, [NSNull null], nil])]);
+  //  self.instanceIDMock = OCMPartialMock([FIRInstanceID instanceIDForTests]);
+  //  FIRInstanceIDResult *result = [[FIRInstanceIDResult alloc] init];
+  //  result.instanceID = @"test-instance-id";
+  //  result.token = @"test-instance-id-token";
+  //  OCMStub([self.instanceIDMock
+  //      instanceIDWithHandler:([OCMArg invokeBlockWithArgs:result, [NSNull null], nil])]);
+
+  // Installations Mock
+  NSString *FID = @"fid-is-better-than-iid";
+  FIRInstallationsAuthTokenResult *FISToken =
+      [[FIRInstallationsAuthTokenResult alloc] initWithToken:@"fake-fis-token" expirationDate:nil];
+  self.installationsMock = OCMPartialMock(
```

Comment:
As mentioned before, it doesn't look like a partial mock is needed here. Could you please replace `OCMPartialMock` with `OCMClassMock` (see the example provided earlier.

Reply:
Yes, done in this commit.